### PR TITLE
Feature val you payement balance

### PR DESCRIPTION
**Pull request by johannschwabe**

### CODE REVIEW

File: Demos/FCFS/ValueFunction/FCFSPathValueFunction.py [hunk 1]
```diff
@@ -5,6 +5,8 @@
 if TYPE_CHECKING:
     from Simulator import PathAgent, Segment
 
+violationValue = - 20
+
```
Comment (thommann) [inline Demos/FCFS/ValueFunction/FCFSPathValueFunction.py:9]: put in constructor to make it parameterizable

File: Demos/FCFS/ValueFunction/FCFSSpaceValueFunction.py [hunk 2]
```diff
@@ -13,15 +13,10 @@ class FCFSSpaceValueFunction(ValueFunction):
...
         sum_blocks = 0.0
         for block in agent.blocks:
-            sum_blocks += (block.max.x - block.min.x) * \
-                          (block.max.y - block.min.y) * \
-                          (block.max.z - block.min.z) * \
-                          (block.max.t - block.min.t)
+            sum_blocks += block.nr_voxels
 
-        return sum_segments / sum_blocks
+        return sum_segments * 0.00006 - (sum_blocks - sum_segments) * 0.0000105
```
Comment (thommann) [inline Demos/FCFS/ValueFunction/FCFSSpaceValueFunction.py:29]: put magic number in constructor and write docstring

File: Demos/Priority/PaymentRule/PriorityPaymentRule.py [hunk 3]
```diff
@@ -16,7 +16,7 @@ class PriorityPaymentRule(PaymentRule):
 
     label = "Priority Payment"
 
-    def __init__(self, path_voxel_multiplier: float = 0.002, space_voxel_multiplier: float = 0.0001):
+    def __init__(self, path_voxel_multiplier: float = 0.2, space_voxel_multiplier: float = 0.00005):
```
Comment (thommann) [inline Demos/Priority/PaymentRule/PriorityPaymentRule.py:19]: explain magic number in docstring

File: Demos/Priority/ValueFunction/PriorityPathValueFunction.py [hunk 4]
```diff
@@ -5,6 +5,8 @@
 if TYPE_CHECKING:
     from Simulator import PathAgent, Segment
 
+violationValue = - 20
+
```
Comment (thommann) [inline Demos/Priority/ValueFunction/PriorityPathValueFunction.py:9]: same

File: Demos/Priority/ValueFunction/PriorityPathValueFunction.py [hunk 5]
```diff
@@ -16,21 +18,26 @@ def value_for_segments(self, segments: List["Segment"], agent: "PathAgent"):
...
-
-        value = 1.
+            return violationValue
+
+        expected_distance = 0
+        _iter = agent.locations[0]
+        for location in agent.locations[1:]:
+            expected_distance += _iter.distance(location) * 1.1
+            _iter = location
+        value = expected_distance * 2 * agent.config["priority"]
```
Comment (thommann) [inline Demos/Priority/ValueFunction/PriorityPathValueFunction.py:35]: why don't we just use the same value function for both allocators?

File: Demos/Priority/ValueFunction/PrioritySpaceValueFunction.py [hunk 6]
```diff
@@ -13,17 +13,13 @@ class PrioritySpaceValueFunction(ValueFunction):
...
-                          (block.max.y - block.min.y) * \
-                          (block.max.z - block.min.z) * \
-                          (block.max.t - block.min.t)
+            sum_blocks += block.nr_voxels
 
         if sum_blocks == 0:
             return 0
-        return sum_segments / sum_blocks
+        return sum_segments * 0.0002 * (1 + agent.config["priority"] / 5) - (
+                (sum_blocks - sum_segments) * agent.config["priority"] * 0.00002 * (1 + agent.config["priority"] / 10))
```
Comment (thommann) [inline Demos/Priority/ValueFunction/PrioritySpaceValueFunction.py:32]: same

**Review by thommann — CHANGES_REQUESTED** [review]
Please provide some comments and docstrings to explain magic numbers

File: Demos/Priority/ValueFunction/PriorityPathValueFunction.py (hunk 5 above)
Comment (johannschwabe) [reply Demos/Priority/ValueFunction/PriorityPathValueFunction.py:35]: agents with a higher priority have a higher value

**Review by thommann — APPROVED** [review]

**Pull request merged**